src/IROperator.cpp

```diff
@@ -446,8 +447,9 @@ Expr lossless_cast(Type t, Expr e) {
     }
 
     if (const Cast *c = e.as<Cast>()) {
-        if (t.can_represent(c->value.type())) {
-            // We can recurse into widening casts.
+        // We can recurse into widening casts.
+        if (c->type.can_represent(c->value.type()) &&
+            t.can_represent(c->value.type())) {
```

Comment:
I think you can get rid of this second part of the condition, and leave that to the recursion?

Reply:
Right